lib/fog/cloudstack/models/compute/image.rb

```diff
@@ -66,6 +66,28 @@ def destroy
           service.delete_template('id' => self.id)
           true
         end
+        def register
+          requires :display_text, :format, :hypervisor, :name, :os_type_id, :url, :zone_id
+          options = {
+                  'displaytext' => display_text,
+                  'format' => format,
+                  'hypervisor' => hypervisor,
+                  'name' => name,
+                  'ostypeid' => os_type_id,
+                  'url' => url,
+                  'zoneid' => zone_id,
+                  'isfeatured' => is_featured
+          }
```

Comment:
Perhaps you can use `all_associations_and_attributes` here as well and D.R.Y it up.